File: csharp/src/Structural/Facade/FacadeWater.cs
namespace Structural.Facade {


//FacadeWater.java - two of three classes the facade calls

public class FacadeWater {  
   bool waterIsBoiling; 
    
   public FacadeWater() {
       setWaterIsBoiling(false);
   }
   
   public void boilFacadeWater() {
       setWaterIsBoiling(true);
   }
   
   public void setWaterIsBoiling(bool isWaterBoiling) {
       waterIsBoiling = isWaterBoiling;
   }
   public bool getWaterIsBoiling() {
       return waterIsBoiling;
   }
}

}

File: scala/src/main/scala/Structural/Facade/FacadeWater.scala
package Structural.Facade;


//FacadeWater - two of three classes the facade calls

class FacadeWater {  
   boolean waterIsBoiling; 
    
   FacadeWater() {
       setWaterIsBoiling(false);
       System.out.println("behold the wonderous water");       
   }
   
   void boilFacadeWater() {
       setWaterIsBoiling(true);
       System.out.println("water is boiling");
   }
   
   void setWaterIsBoiling(boolean isWaterBoiling) {
       waterIsBoiling = isWaterBoiling;
   }
   boolean getWaterIsBoiling() {
       return waterIsBoiling;
   }
}